### PR TITLE
feat: support Windows

### LINKED ISSUE #76: Create binary packages for Windows
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

Install fails on Windows with:

```
Encountered problems while solving:
  - nothing provides requested google-cloud-cpp >=1.34.0
```

I would be happy to submit a PR if you give me some pointers to tutorials or documentation on how to do this.

https://ci.appveyor.com/project/ApacheSoftwareFoundation/arrow/builds/41764812/job/i8knrq50fpnmfhc0

<br/>
Environment (<code>conda list</code>):

> Sorry, not available from the AppVeyor logs

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
C:\projects\arrow>conda info -a 
     active environment : None
       user config file : C:\Users\appveyor\.condarc
 populated config files : C:\Users\appveyor\.condarc
          conda version : 4.7.12
    conda-build version : not installed
         python version : 3.7.4.final.0
       virtual packages : 
       base environment : C:\Miniconda37-x64  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Miniconda37-x64\pkgs
                          C:\Users\appveyor\.conda\pkgs
                          C:\Users\appveyor\AppData\Local\conda\conda\pkgs
       envs directories : C:\Miniconda37-x64\envs
                          C:\Users\appveyor\.conda\envs
                          C:\Users\appveyor\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.7.12 requests/2.22.0 CPython/3.7.4 Windows/10 Windows/10.0.14393
          administrator : True
             netrc file : None
           offline mode : False
# conda environments:
#
base                  *  C:\Miniconda37-x64
sys.version: 3.7.4 (default, Aug  9 2019, 18:34:13) [...
sys.prefix: C:\Miniconda37-x64
sys.executable: C:\Miniconda37-x64\python.exe
conda location: C:\Miniconda37-x64\lib\site-packages\conda
conda-build: None
conda-env: C:\Miniconda37-x64\Scripts\conda-env.exe
user site dirs: 
APR_ICONV_PATH: C:\Program Files (x86)\Subversion\iconv
CIO_TEST: <not set>
CONDA_ROOT: C:\Miniconda37-x64
GIT_LFS_PATH: C:\Program Files\Git LFS
HOMEPATH: \Users\appveyor
PATH: C:\Miniconda37-x64;C:\Miniconda37-x64\Scripts;C:\Miniconda37-x64\Library\bin;C:\Program Files\Docker\Docker\Resources\bin;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\Perl\site\bin;C:\Perl\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Program Files\7-Zip;C:\Tools\GitVersion;C:\Tools\NuGet;C:\Program Files\Microsoft\Web Platform Installer\;C:\Tools\PsTools;C:\Program Files\Git LFS;C:\Program Files\Mercurial\;C:\Program Files (x86)\Subversion\bin;C:\Tools\WebDriver;C:\Tools\Coverity\bin;C:\Tools\MSpec;C:\Tools\NUnit\bin;C:\Tools\NUnit3;C:\Tools\xUnit;C:\Program Files\nodejs;C:\Program Files (x86)\iojs;C:\Program Files\iojs;C:\Program Files\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\110\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\;C:\Program Files\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\Binn\ManagementStudio\;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files\Microsoft SQL Server\130\Tools\Binn\;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files\Microsoft SQL Server\130\DTS\Binn\;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\130\Tools\Binn\;C:\Ruby193\bin;C:\go\bin;C:\Program Files\Java\jdk1.8.0\bin;C:\Program Files (x86)\Apache\Maven\bin;C:\Python27;C:\Python27\Scripts;C:\Program Files (x86)\CMake\bin;C:\Tools\curl\bin;C:\Program Files\Microsoft DNX\Dnvm\;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\MSBuild\15.0\Bin;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\Extensions\Microsoft\SQLDB\DAC\130;C:\Program Files\dotnet\;C:\Tools\vcpkg;C:\Program Files (x86)\dotnet\;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\Binn\;C:\Program Files\Microsoft SQL Server\140\Tools\Binn\;C:\Program Files\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\IDE\Extensions\TestPlatform;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn\;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn\;C:\Program Files\erl9.2\bin;C:\Program Files (x86)\NSIS;C:\Tools\Octopus;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\Docker\Docker\resources;C:\Program Files\LLVM\bin;C:\Users\appveyor\AppData\Roaming\npm;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\;C:\Program Files\PowerShell\6\;C:\Program Files (x86)\nodejs\;C:\Program Files\Git\cmd;C:\Program Files\Git\usr\bin;C:\Program Files\Meson\;C:\ProgramData\chocolatey\bin;C:\Program Files\Amazon\AWSCLI\;C:\Program Files (x86)\Yarn\bin\;C:\Users\appveyor\AppData\Local\Microsoft\WindowsApps;C:\Users\appveyor\.dotnet\tools;C:\Users\appveyor\AppData\Roaming\npm;C:\Users\appveyor\AppData\Local\Yarn\bin;C:\Program Files\AppVeyor\BuildAgent\
PSMODULEPATH: C:\Users\appveyor\Documents\WindowsPowerShell\Modules;C:\Program Files\WindowsPowerShell\Modules;C:\Windows\system32\WindowsPowerShell\v1.0\Modules;C:\Users\appveyor\Documents\WindowsPowerShell\Modules;C:\Program Files (x86)\Microsoft SQL Server\120\Tools\PowerShell\Modules;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\PowerShell\Modules;C:\Program Files (x86)\AWS Tools\PowerShell;C:\Program Files\AppVeyor\BuildAgent\Modules;C:\Program Files (x86)\Microsoft SQL Server\140\Tools\PowerShell\Modules;C:\Program Files (x86)\Microsoft SDKs\Azure\PowerShell\ResourceManager\AzureResourceManager;C:\Program Files (x86)\Microsoft SDKs\Azure\PowerShell\ServiceManagement;C:\Program Files (x86)\Microsoft SDKs\Azure\PowerShell\Storage;C:\Program Files (x86)\Microsoft SQL Server\130\Tools\PowerShell\Modules\
PYTHON: 3.7
REQUESTS_CA_BUNDLE: <not set>
SSL_CERT_FILE: <not set>

```
</details>

### PR DESCRIPTION
**Pull request by coryan**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* N/A Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #76

### CODE REVIEW

**Comment by coryan** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -23,6 +24,7 @@ requirements:
     - ninja
     - grpc-cpp
     - libprotobuf
+    - git
```
Comment (xhochy) [inline recipe/meta.yaml:27]: This shouldn't be needed 🤔

File: recipe/run_test.bat [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+set LIBRARY_PREFIX="%LIBRARY_PREFIX:\=/%"
+
+cmake -GNinja ^
+    -S google/cloud/storage/quickstart -B .build/quickstart ^
+    -DCMAKE_CXX_STANDARD=17 ^
+    -DCMAKE_BUILD_TYPE=Release ^
+    -DCMAKE_PREFIX_PATH="%LIBRARY_PREFIX%" ^
+    -DCMAKE_MODULE_PATH="%LIBRARY_PREFIX%/lib/cmake"
+
+cmake --build .build/quickstart --config Release
```
Comment (xhochy) [inline recipe/run_test.bat:14]:
```suggestion
cmake -GNinja ^
    -S google/cloud/storage/quickstart -B .build/quickstart ^
    -DCMAKE_CXX_STANDARD=17 ^
    -DCMAKE_BUILD_TYPE=Release ^
    -DCMAKE_PREFIX_PATH="%LIBRARY_PREFIX%" ^
    -DCMAKE_MODULE_PATH="%LIBRARY_PREFIX%/lib/cmake"
if NOT ERRORLEVEL 0 exit /b 1

cmake --build .build/quickstart --config Release
if NOT ERRORLEVEL 0 exit /b 1
```

File: recipe/meta.yaml (hunk 1 above)
Comment (coryan) [reply recipe/meta.yaml]: Removed.  Maybe my setup was broken, but testing locally  on Windows failed without it.  :shrug:  We can always add it back if something fails.

**Comment by xhochy** [comment]
@conda-forge-admin please rerender

**Comment by xhochy** [comment]
@coryan feel free to add yourself as a maintainer here to also be able to merge things :)

**Comment by coryan** [comment]
Good idea, sending #79.  I assume that will change the GitHub permissions?

**Comment by xhochy** [comment]
@coryan feel free to add yourself as a maintainer here to also be able to merge things :)

**Comment by xhochy** [comment]
> Good idea, sending #79. I assume that will change the GitHub permissions?

Yes!

@conda-forge-admin please rerender

**Pull request merged**